r/tests/testthat/test-Array.R

```diff
@@ -260,11 +260,10 @@ test_that("array supports POSIXct (ARROW-3340)", {
   expect_array_roundtrip(times2, timestamp("us", "US/Eastern"))
 })
 
-test_that("array supports POSIXct without timezone", {
-  # Make sure timezone is not set
-  withr::with_envvar(c(TZ = ""), {
+test_that("array uses local timezone for POSIXct without timezone", {
+  withr::with_envvar(c(TZ = "Asia/Ulaanbaatar"), {
```

Comment:
I think we want to keep this test with `withr::with_envvar(c(TZ = ""), {` like it is (though we will probably need to change the line below to be something like `expect_array_roundtrip(times, timestamp("us", Sys.timezone()))`.

Additionally, we should have a test that is similar to this one that _also_ sets `withr::with_envvar(c(TZ = NA), {` since that is most like the environments where someone has not actually proactively changed any of their timezone settings.

Reply:
Done